src/os/feature/dynamicfeature.js

```diff
@@ -0,0 +1,92 @@
+goog.provide('os.feature.DynamicFeature');
+goog.provide('os.feature.DynamicPropertyChange');
+
+goog.require('ol.Feature');
+
+
+/**
+ * Dynamic feature property change types.
+ * @enum {string}
+ */
+os.feature.DynamicPropertyChange = {
+  GEOMETRY: 'change:originalGeometry'
+};
+
+
+/**
+ * A dynamic feature that changes with time.
+ * @param {ol.geom.Geometry|Object<string, *>=} opt_geometryOrProperties
+ *     You may pass a Geometry object directly, or an object literal
+ *     containing properties.  If you pass an object literal, you may
+ *     include a Geometry associated with a `geometry` key.
+ * @param {function(!ol.Feature)=} opt_initFn Initialize the feature into the animating state.
+ * @param {function(!ol.Feature, boolean=)=} opt_disposeFn Restore the feature to the non-animating state.
+ * @param {function(!ol.Feature, number)=} opt_updateFn Update the animating state for the given timestamp.
+ * @extends {ol.Feature}
+ * @constructor
+ */
+os.feature.DynamicFeature = function(opt_geometryOrProperties, opt_initFn, opt_disposeFn, opt_updateFn) {
+  os.feature.DynamicFeature.base(this, 'constructor', opt_geometryOrProperties);
+
+  /**
+   * Initialize the feature into the animating state.
+   * @type {function(!ol.Feature)}
+   */
+  this.initFn = opt_initFn || goog.nullFunction;
+
+  /**
+   * Restore the feature to the non-animating state.
+   * @type {function(!ol.Feature, boolean=)}
+   */
+  this.disposeFn = opt_disposeFn || goog.nullFunction;
+
+  /**
+   * Update the animating state for the given timestamp.
+   * @type {function(!ol.Feature, number)}
+   */
+  this.updateFn = opt_updateFn || goog.nullFunction;
+};
+goog.inherits(os.feature.DynamicFeature, ol.Feature);
+
+
+/**
+ * Updates the feature to display for the given timestamp.
+ */
+os.feature.DynamicFeature.prototype.initDynamic = function() {
+  this.initFn(this);
+};
+
+
+/**
+ * Updates the feature to display for the given timestamp.
+ * @param {boolean=} opt_disposing If the feature is being disposed.
+ */
+os.feature.DynamicFeature.prototype.disposeDynamic = function(opt_disposing) {
+  this.disposeFn(this, opt_disposing);
+};
+
+
+/**
+ * Updates the feature to display for the given timestamp.
+ * @param {number} timestamp The timestamp.
+ */
+os.feature.DynamicFeature.prototype.updateDynamic = function(timestamp) {
+  this.updateFn(this, timestamp);
+};
+
+
+/**
+ * @inheritDoc
+ * @suppress {accessControls} To allow direct access to feature metadata.
+ */
+os.feature.DynamicFeature.prototype.clone = function() {
+  var clone = new os.feature.DynamicFeature(undefined, this.initFn, this.disposeFn, this.updateFn);
+  clone.setProperties(this.values_, true);
```

Comment:
This could be done with a `getProperties()` to avoid the suppress tag.

Reply:
It's done this way for performance reasons, to avoid an unnecessary copy of the properties. We did the same thing in our mixin for `ol.Feature#clone`.

Reply:
Gotcha. Figured this wasn't called that much so the usual performance caveat wouldn't apply.